### PR TITLE
Upgrade to Ruby 2.7.4

### PR DESCRIPTION
**Pull request by ekowidianto**
**References:**
https://www.ruby-lang.org/en/news/2019/12/25/ruby-2-7-0-released/
https://rubyreferences.github.io/rubychanges/2.7.html

**Main breaking change (Warning in 2.7 and error in 3.0):** 
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

**Forked and updated gems:**
activerecord-userstamp: https://github.com/ekowidianto/activerecord-userstamp/commit/1170233e314466630e3a545ae3fdee16a0bb4a19
active_record-acts_as: https://github.com/ekowidianto/active_record-acts_as/commit/9e01feb01438ba6736f2d4a8140d9be18464afdc

**Note:**
deprecation warnings are turned off by default on 2.7.2 and later. Use RUBYOPT='-W:deprecated' to enable warnings when **testing locally.**

### CODE REVIEW

File: app/jobs/course/assessment/submission/auto_grading_job.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Course::Assessment::Submission::AutoGradingJob < ApplicationJob
   #   results into.
   # @param [Boolean] only_ungraded Whether grading should be done ONLY for
   #   ungraded_answers, or for all answers regardless of workflow state
-  def perform_tracked(submission, only_ungraded: false)
+  def perform_tracked(submission, only_ungraded = false) # rubocop:disable Style/OptionalBooleanParameter
```
Comment (zhuhanming) [inline app/jobs/course/assessment/submission/auto_grading_job.rb:14]: Any reason for doing `=` instead of `:`? Just curious.

**Review by zhuhanming — APPROVED** [review]
LGTM! Just one small question below.

File: app/jobs/course/assessment/submission/auto_grading_job.rb (hunk 1 above)
Comment (ekowidianto) [reply app/jobs/course/assessment/submission/auto_grading_job.rb:14]: In Rails API, the perform_later method takes in *args as the only argument. As such, if we pass in a keyword argument here (https://github.com/Coursemology/coursemology2/pull/4165/files#diff-11bda8445480301b60c1f569535d4c5f6c3aa44819fa7b2cecd08fbdd4f73591R180), it cant be properly passed to the perform_tracked method based on the new ruby 2.7 kwargs update.

Comment (ekowidianto) [reply app/jobs/course/assessment/submission/auto_grading_job.rb:14]: So if we want to use 
`def perform_tracked(submission, only_ungraded: false)`,
we need to pass
AutoGradingJob.perform_later([self, only_ungraded: only_ungraded])

Comment (zhuhanming) [reply app/jobs/course/assessment/submission/auto_grading_job.rb:14]: I see, so there's some difference between an optional param vs a keyword param with a default value, and it seems like Rubocop prefers the latter?

Comment (ekowidianto) [reply app/jobs/course/assessment/submission/auto_grading_job.rb:14]: for boolean only, yes rubocop prefers the later

**Pull request merged**